packages/sns/src/ledger.canister.ts

```diff
@@ -40,4 +43,28 @@ export class SnsLedgerCanister extends Canister<SnsLedgerService> {
       owner: params.owner,
       subaccount: toNullable(params.subaccount),
     });
+
+  /**
+   * Transfers tokens from the sender to the given account.
+   *
+   * @param {TransferArg} params The parameters to transfer tokens.
+   *
+   * @throws {SnsTransferError} If the transfer fails.
+   */
+  transfer = async (params: TransferParams): Promise<BlockIndex> => {
+    const response = await this.caller({ certified: true }).icrc1_transfer(
+      toTransferArg(params)
+    );
+    if ("Err" in response) {
+      throw new SnsTransferError({
+        type: response.Err,
+        msg: "Failed to transfer",
+      });
+    }
+    if ("Ok" in response) {
+      return response.Ok;
+    }
+    // Edger case, should never happen.
+    throw new Error("Unexpected response from transfer");
```

Comment:
isn't the `response` a variant? it can be something else than error or ok?

Reply:
It's a variant with `Ok` and `Error` only.

Reply:
```
type TransferResult = variant {
    Ok : BlockIndex;
    Err : TransferError;
};
```

Reply:
Is there an edge case then, should we simplify it?

```
if ("Err" in response) {
      throw new SnsTransferError({
        type: response.Err,
        msg: "Failed to transfer",
      });
    }
    
return response.Ok;
```

In Motoko at least the Result type (https://internetcomputer.org/docs/current/references/motoko-ref/result/) is ok or err and cannot be extended I think.

Reply:
Sure

Reply:
done